client/starwhale/core/runtime/model.py

```diff
@@ -1963,11 +1967,13 @@ def restore(
 
         _manifest = load_yaml(workdir / DEFAULT_MANIFEST_NAME)
         _env = _manifest["environment"]
-        _starwhale_version = (
-            _manifest.get("environment", {})
-            .get("lock", {})
-            .get("starwhale_version", "")
+        _lock = _manifest.get("environment", {}).get("lock", {})
+        expected_pyver = (
+            _env["python"]
+            or _lock.get("shell", {}).get("python_version")
+            or get_python_version()
```

Comment:
What is the difference between `environment.python` and `environment.lock.python_version`?
(It seems that we usually use the variable that does not contain the minor version number)

Reply:
`environment.python` --> defined by the users in the runtime.yaml, the field is optional.
`environment.lock.shell.python_version` --> `swcli runtime build` command uses python bin version that is locked automatically. 

Currently, the 'Starwhale Runtime' only supports Python version formats in the 'major.minor' pattern, such as '3.7' and '3.8'. The 'micro' field will be omitted.